web/viewer.js

```diff
@@ -1061,9 +1062,13 @@ var PDFViewerApplication = {
     });
   },
 
-  setInitialView: function pdfViewSetInitialView(storedHash, scale) {
+  setInitialView: function pdfViewSetInitialView(storedHash, scale, rotation) {
     this.isInitialViewSet = true;
 
+    if (rotation && rotation !== this.pageRotation) {
```

Comment:
I don't think this works as intended, since it will disallow `rotation === 0`.
If you want to check that `rotation` is an integer, perhaps do `(rotation | 0) === rotation` instead?

Also, do we trust the value we get from storage, or should it be validated further. e.g. by checking that it's a multiple of `90`?